### PR TITLE
8276227: ciReplay: SIGSEGV if classfile for replay compilation is not present after JDK-8275868

### PR DESCRIPTION
**Pull request by GoeLin**
I had to do a trivial resolve in ciReplay and adapt the path to WhiteBox in the test.

Follow up for 8275868.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8276227](https://bugs.openjdk.org/browse/JDK-8276227) needs maintainer approval

### Issue
 * [JDK-8276227](https://bugs.openjdk.org/browse/JDK-8276227): ciReplay: SIGSEGV if classfile for replay compilation is not present after JDK-8275868 (**Bug** - P3 - Approved)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [83ace890](https://git.openjdk.org/jdk17u-dev/pull/2335/files/83ace890844293f5ac86d5f3f254263005e96aa4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2335/head:pull/2335` \
`$ git checkout pull/2335`

Update a local copy of the PR: \
`$ git checkout pull/2335` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2335/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2335`

View PR using the GUI difftool: \
`$ git pr show -t 2335`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2335.diff">https://git.openjdk.org/jdk17u-dev/pull/2335.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2335#issuecomment-2022280355)

### CODE REVIEW

File: src/hotspot/share/ci/ciReplay.cpp [hunk 1]
```diff
@@ -685,12 +687,17 @@ class CompileReplay : public StackObj {
...
     // just load the referenced class
     Klass* k = parse_klass(CHECK);
-    if (_protection_domain() == NULL) {
+    if (!_protection_domain_initialized && k != NULL) {
+      assert(_protection_domain() == NULL, "must be uninitialized");
       // The first entry is the holder class of the method for which a replay compilation is requested.
       // Use the same protection domain to load all subsequent classes in order to resolve all classes
       // in signatures of inlinees. This ensures that inlining can be done as stated in the replay file.
       _protection_domain = Handle(_thread, k->protection_domain());
     }
```
Comment (TheRealMDoerr) [inline src/hotspot/share/ci/ciReplay.cpp:696]: Original change inserts an empty line, here.

**Review by TheRealMDoerr — APPROVED** [review]
LGTM. Please consider inserting the empty line before integrating.

File: src/hotspot/share/ci/ciReplay.cpp (hunk 1 above)
Comment (GoeLin) [reply src/hotspot/share/ci/ciReplay.cpp:696]: Fixed.

**Pull request closed without merging**